src/python/pants/backend/jvm/tasks/ivy_task_mixin.py

```diff
@@ -101,15 +101,32 @@ def implementation_version(cls):
     return super(IvyTaskMixin, cls).implementation_version() + [('IvyTaskMixin', 4)]
 
   @memoized_property
-  def ivy_cache_dir(self):
-    """The path of the ivy cache dir used for resolves.
+  def ivy_repository_cache_dir(self):
+    """The path of the ivy cache dir used for repository cache data.
 
     :API: public
 
     :rtype: string
     """
     # TODO(John Sirois): Fixup the IvySubsystem to encapsulate its properties.
-    return IvySubsystem.global_instance().get_options().cache_dir
+    if IvySubsystem.global_instance().get_options().repository_cache_dir:
+      return IvySubsystem.global_instance().get_options().repository_cache_dir
+    else:
+      return IvySubsystem.global_instance().get_options().cache_dir
+
+  @memoized_property
+  def ivy_resolution_cache_dir(self):
+    """The path of the ivy cache dir used for resolution cache data.
+
+    :API: public
+
+    :rtype: string
+    """
+    # TODO(John Sirois): Fixup the IvySubsystem to encapsulate its properties.
+    if IvySubsystem.global_instance().get_options().resolution_cache_dir:
```

Comment:
Rather than checking for presence, I'd recommend giving both of the new options default values, and then checking to see whether the cache_dir was set explicitly:
```
...get_options().is_default('cache_dir')
```

So approximately:
```
if IvySubsystem.global_instance().get_options().is_default('cache_dir'):
  # use resolution/repository options
else:
  # use cache_dir option
```